client/src/components/Traveller.jsx

```diff
@@ -8,8 +8,17 @@ import CommentBox from './reusable/CommentBox';
 import ImageBox from './reusable/ImageBox';
 import { dateToStr, dateTimeToStr } from '../helpers/helpers';
 import * as Constants from './Constants';
+import { AuthContext } from './AuthContext';
+import ConfirmBox from './reusable/ConfirmBox';
 
-class Traveller extends Component {
+const Traveller = (props) => {
+  const authData = useContext(AuthContext);
+  return (
+    <TravellerWithAuth {...props} userData={authData.authProviderMounted && authData.isAuth ? authData : null} />
+  );
+};
+
+class TravellerWithAuth extends Component {
```

Comment:
@tom-gregorovic hmmm, teraz som pochopil ze tu vlastne dealujes s technickym dlhom v zmysle ze stare komponenty su napisane cez `class XXX extends` a tym padom neberu pouzitie `React.context` takze ich balis to funkcneho komponentu aby si si vedel poslat informacie o `auth` cez `props`. je to celkom clever riesenie vzniknutej situacie a urcite rychlejsie aplikovatelne, ale ak by sa ti chcelo, idealny pristup by bol zmenit ten `class component` na `const` compoenent, namiesto `constructora` by sa pouzil len `useState` z Reactu. daj vediet ci sa ti do toho chce, malo by to benefit v tom ze by to bola `best practice`, a bolo by to citatelnejsie do buducnosti. ak si za, kludne mozme dat pair programing a mozme jeden komponent prerobit spolu

Reply:
Zkusím ty nové věci upravit tímhle způsobem. Existující funkční kód bych za každou cenu nepřepisoval. Když to funguje, radši do toho nešahám. Navíc by mi tak velké úpravy přidaly spoustu konfliktů, protože vyvíjím souběžně odděleně více změn.